src/java-interop/java-interop.csproj

```diff
@@ -50,20 +42,83 @@
+      DependsOnTargets="BuildJni_c"
+      Inputs="@(ClCompile)"
+      Outputs="obj\$(Configuration)\%(Filename).o">
+    <MakeDir Directories="obj\$(Configuration)" />
+    <ItemGroup>
+      <_Defines Include="%(ClCompile.PreprocessorDefinitions)" />
+      <_Includes Include="%(ClCompile.AdditionalIncludeDirectories)" />
+    </ItemGroup>
+    <PropertyGroup>
+      <_Arch Condition=" Exists ('/Library/Frameworks/') ">-m64</_Arch>
+      <_Cc Condition=" '%(ClCompile.Extension)' == '.c' ">gcc -std=c99 -fPIC</_Cc>
+      <_Cc Condition=" '%(ClCompile.Extension)' == '.cc' ">g++ -std=c++11 -fPIC</_Cc>
+      <_Def>@(_Defines->'-D%(Identity)', ' ')</_Def>
+      <_Inc>@(_Includes->'-I "%(Identity)"', ' ')</_Inc>
+    </PropertyGroup>
+    <Exec Command="$(_Cc) -c -g $(_Arch) -o &quot;obj\$(Configuration)\%(ClCompile.Filename).o&quot; $(_Def) $(_Inc) &quot;%(Identity)&quot;" />
+  </Target>
+
+  <Target Name="BuildMac" AfterTargets="Build"
+      Condition=" '$(OS)' != 'Windows_NT' And Exists ('/Library/Frameworks/')"
+      DependsOnTargets="_CompileObjectFiles"
+      Inputs="@(ClCompile)"
+      Outputs="$(_MacLib)">
+    <PropertyGroup>
+      <_LinkFlags>-fvisibility=hidden -Wl,-undefined -Wl,suppress -Wl,-flat_namespace</_LinkFlags>
+      <_Libs>$(MonoLibs)</_Libs>
+      <_Files>@(ClCompile->'obj\$(Configuration)\%(Filename).o', ' ')</_Files>
+    </PropertyGroup>
+    <Exec Command="gcc -g -shared -m64 -std=c99 -fPIC -o &quot;$(_MacLib)&quot; $(_LinkFlags) $(_Libs) $(_Files)" />
+    <!-- Mono 4.4.0 (mono-4.4.0-branch/a3fabf1) has an incorrect shared library name. Fix it -->
+    <Exec Command="install_name_tool -change /private/tmp/source-mono-4.4.0/bockbuild-mono-4.4.0-branch/profiles/mono-mac-xamarin/package-root/lib/libmonosgen-2.0.1.dylib /Library/Frameworks/Mono.framework/Libraries/libmonosgen-2.0.1.dylib &quot;$(_MacLib)&quot;" />
+  </Target>
+
+  <Target Name="BuildUnixLibraries" AfterTargets="Build"
+      Condition=" '$(OS)' != 'Windows_NT' And !Exists ('/Library/Frameworks/')"
+      DependsOnTargets="_CompileObjectFiles"
+      Inputs="@(ClCompile)"
+      Outputs="$(OutputPath)\lib$(OutputName).so">
+    <PropertyGroup>
+      <_FixedDefines>$(DefineSymbols.Split(' '))</_FixedDefines>
+    </PropertyGroup>
+    <ItemGroup>
+      <_Defines Include="$(_FixedDefines)" />
+    </ItemGroup>
+    <PropertyGroup>
+      <_LinkFlags>-fvisibility=hidden -Wl,-undefined -Wl,suppress -Wl,-flat_namespace -fPIC</_LinkFlags>
+      <_Libs>$(MonoLibs)</_Libs>
+      <_Files>@(ClCompile->'obj\$(Configuration)\%(Filename).o', ' ')</_Files>
+    </PropertyGroup>
+    <Exec Command="g++ -g -shared -o &quot;$(OutputPath)\lib$(OutputName).so&quot; $(_LinkFlags) $(_Libs) $(_Files)" />
+  </Target>
+
+  <Target Name="Clean">
+    <RemoveDir Directories="obj" />
+    <Delete Files="jni.c" />
+    <Delete
+        Files="$(OutputPath)\lib$(OutputName).dylib"
+        Condition=" '$(OS)' != 'Windows_NT' "
+    />
+  </Target>
```

Comment:
I think the reason these targets were in a separate file, is because VS will auto-format the file in weird ways.

Is this not an issue anymore with SDK-style projects?

Reply:
🤷 Must not be, I've removed most of the `.targets` files in this repo as I move them to SDK style, and I use VS exclusively.

Reply:
It might be VS for Mac.

Reply:
If we're going Modern, these targets should be in `Directory.Build.targets`.

Additionally, it *is* Xamarin Studio (née Visual Studio for Mac) which is responsible for reformatting files and otherwise breaking things, e.g.

  * https://xamarin.github.io/bugzilla-archives/42/42603/bug.html
  * https://xamarin.github.io/bugzilla-archives/14/14004/bug.html

I can't find a bug specific to reformatting `.csproj` files and losing whitespace changes, but the above two issues deal with reformatting `.csproj` files and otherwise *breaking* things…

Reply:
I tested in VSMac and it does indeed correct our crazy indentation: https://gist.github.com/jpobst/9473335ecd4d3fe4e7bdfeb30dcee9ac.  I'll move it back to `java-interop.targets` so we can keep our insanity even if someone ever uses VSMac. 😉

Reply:
Updated.